sdk/program/src/fee_calculator.rs

```diff
@@ -32,20 +33,22 @@ impl FeeCalculator {
         note = "Please do not use, will no longer be available in the future"
     )]
     pub fn calculate_fee(&self, message: &Message) -> u64 {
-        let mut num_secp256k1_signatures: u64 = 0;
+        let mut num_signatures: u64 = 0;
         for instruction in &message.instructions {
             let program_index = instruction.program_id_index as usize;
             // Message may not be sanitized here
             if program_index < message.account_keys.len() {
                 let id = message.account_keys[program_index];
-                if secp256k1_program::check_id(&id) && !instruction.data.is_empty() {
-                    num_secp256k1_signatures += instruction.data[0] as u64;
+                if (secp256k1_program::check_id(&id) || ed25519_program::check_id(&id))
+                    && !instruction.data.is_empty()
+                {
+                    num_signatures += instruction.data[0] as u64;
```

Comment:
This is at best confusing since the count is represented as a u16.  Might be more straightforward and simple to go back to a u8 count as the first byte, a padding byte, and then the bytemuck serialized data?  Unless you think a u8 would overflow at some point in the future

Reply:
Good catch.

Reply:
Fixed. I've added a single padding byte.